modules/docker/host.go

```diff
@@ -0,0 +1,40 @@
+package docker
+
+import (
+	"os"
+	"strings"
+)
+
+// GetDockerHost returns the name or address of the host on which the Docker engine is running.
+func GetDockerHost() string {
+	return getDockerHostFromEnv(os.Environ())
+}
+
+func getDockerHostFromEnv(env []string) string {
+	// Parses the DOCKER_HOST environment variable to find the address
+	//
+	// For valid formats see:
+	// https://github.com/docker/cli/blob/6916b427a0b07e8581d121967633235ced6db9a1/opts/hosts.go#L69
+	var dockerUrl []string
+
+	for _, item := range env {
+		envVar := strings.Split(item, "=")
+		if envVar[0] == "DOCKER_HOST" {
+			dockerUrl = strings.Split(envVar[1], ":")
+			break
+		}
+	}
+
+	if len(dockerUrl) == 0 {
+		// no DOCKER_HOST variable, return default value
+		return "localhost"
+	}
+
+	switch dockerUrl[0] {
+	case "tcp", "ssh", "fd":
+		return strings.TrimPrefix(dockerUrl[1], "//")
```

Comment:
Need to check `len(dockerUrl) > 1`

Reply:
I know you're checking that `len(dockerUrl)` above is not 0, but that still doesn't mean there are at least two items here, so we need that check somewhere...

Reply:
This is checked on line 24 if/when we apply `strings.Split` - but could be clearer, I'll refactor it to a single check. Thanks!